os/drivers/serial/serial.c

```diff
@@ -272,7 +273,10 @@ static int uart_putxmitchar(FAR uart_dev_t *dev, int ch, bool oktoblock)
 			/* only save UART data that is being transmitted to the console */
 
 			if (dev->isconsole) {
-				log_dump_save(ch);
+				struct log_dump_msg_s msg;
+				msg.comp = false;
+				msg.ch = ch;
+				mq_send(log_dump_get_mqfd(), (const char *)&msg, sizeof(struct log_dump_msg_s), 100);
```

Comment:
Sending a message for each character will cause tooooo many redundant work. I suggest that 
1. keep original to save a character
2. When a node is full, then send message to compress.

How do you think?